public/app/features/variables/state/sharedReducer.test.ts

```diff
@@ -16,38 +15,78 @@ import {
   variableStateFetching,
   variableStateNotStarted,
 } from './sharedReducer';
-import { QueryVariableModel, VariableHide } from '../types';
+import { ConstantVariableModel, QueryVariableModel, VariableHide, VariableOption } from '../types';
 import { ALL_VARIABLE_TEXT, ALL_VARIABLE_VALUE, toVariablePayload } from './types';
 import { variableAdapters } from '../adapters';
 import { createQueryVariableAdapter } from '../query/adapter';
 import { initialQueryVariableModelState } from '../query/reducer';
 import { getVariableState, getVariableTestContext } from './helpers';
 import { initialVariablesState, VariablesState } from './variablesReducer';
 import { changeVariableNameSucceeded } from '../editor/reducer';
+import { createConstantVariableAdapter } from '../constant/adapter';
+import { initialConstantVariableModelState } from '../constant/reducer';
 
-variableAdapters.setInit(() => [createQueryVariableAdapter()]);
+variableAdapters.setInit(() => [createQueryVariableAdapter(), createConstantVariableAdapter()]);
 
 describe('sharedReducer', () => {
   describe('when addVariable is dispatched', () => {
     it('then state should be correct', () => {
-      const model = ({
+      const model: any = {
         name: 'name from model',
         type: 'type from model',
         current: undefined,
-      } as unknown) as QueryVariableModel;
+      };
 
-      const payload = toVariablePayload({ id: '0', type: 'query' }, { global: true, index: 0, model });
+      const expected: QueryVariableModel = {
+        ...initialQueryVariableModelState,
+        id: 'name from model',
+        global: true,
+        index: 0,
+        name: 'name from model',
+        type: ('type from model' as unknown) as VariableType,
+        current: ({} as unknown) as VariableOption,
+      };
+
+      const payload = toVariablePayload({ id: 'name from model', type: 'query' }, { global: true, index: 0, model });
 
       reducerTester<VariablesState>()
         .givenReducer(sharedReducer, { ...initialVariablesState })
         .whenActionIsDispatched(addVariable(payload))
         .thenStateShouldEqual({
-          [0]: {
-            ...lodashDefaults({}, model, initialQueryVariableModelState),
```

Comment:
I changed this test because the expect is actually implementing the reducer logic.